### PR TITLE
task/message_list/master_fragment

### PR DESCRIPTION
**Pull request by teamagam1**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/55" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/DiskLogger.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ private void logToFile(String verbosity, String message) {
         try {
             mLogWriter.write(completeLogLine);
         } catch (IOException e) {
-            throw new LogWriteException();
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/DiskLogger.java]: That doesn't seems like the solution.....
At least use the native logger to log it, if not understand if there's anything else we can do (retry X times?) !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/FriendsActivity.java [hunk 2]
```diff
@@ -1,57 +0,0 @@
-package com.teamagam.gimelgimel.app.view;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/FriendsActivity.java:1]: Hooray !

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            mRecyclerView.setLayoutManager(new LinearLayoutManager(context));
+            mMessagesViewModel = new MessageListViewModel(this);
+            mAdapter = new MessagesRecyclerViewAdapter(mMessagesViewModel.getRandomAccessor(), this);
+            mRecyclerView.setAdapter(mAdapter);
+        }
+        return view;
+    }
+
+    @Override
+    public void onAttach(Context context) {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java]: remove !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_messages_master_list, container, false);
+
+
+        // Set the adapter
+        if (view instanceof RecyclerView) {
+            Context context = view.getContext();
+            mRecyclerView = (RecyclerView) view;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java]: Get RecyclerView with butterknife using its id !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+     * This interface must be implemented by activities that contain this
+     * fragment to allow an interaction in this fragment to be communicated
+     * to the activity and potentially other fragments contained in that
+     * activity.
+     * <p/>
+     * See the Android Training lesson <a href=
+     * "http://developer.android.com/training/basics/fragments/communicating.html"
+     * >Communicating with Other Fragments</a> for more information.
+     */
+    public interface OnMessageMasterFragmentClickListener {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java]: If we're to do MVVM, there wont be any direct interaction between the Fragment and its container. Right ?

File: app/src/main/res/drawable/ic_message_types.xml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (BarAgam) [inline app/src/main/res/drawable/ic_message_types.xml]: This drawable is empty. Is it on purpose?

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -44,12 +44,18 @@
...
+            <!--android:layout_height="wrap_content"-->
+            <!--android:layout_alignParentBottom="true"-->
+            <!--/>-->
         <FrameLayout
-            android:id="@+id/container_footer"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_alignParentBottom="true"
-            />
+        android:id="@+id/container_footer"
```
Comment (BarAgam) [inline app/src/main/res/layout/activity_main.xml]: Indentation..., !fix

File: app/src/main/res/values/dimens.xml [hunk 8]
```diff
@@ -20,5 +20,6 @@
     <!-- fabs -->
     <dimen name="fab_big">56dp</dimen>
     <dimen name="fab_normal">76dp</dimen>
+    <dimen name="text_margin">16dp</dimen>
```
Comment (BarAgam) [inline app/src/main/res/values/dimens.xml]: misplaced - under the <--- fab --> section.
Probably should be elsewhere... !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/common/logging/DiskLogger.java (hunk 1 above)
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/logging/DiskLogger.java]: I did it myself. Code is already in dev. !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java (hunk 5 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java]: that's right. lets wait for integration

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java (hunk 4 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java (hunk 3 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java]: !resolved

File: app/src/main/res/drawable/ic_message_types.xml (hunk 6 above)
Comment (teamagam1) [inline app/src/main/res/drawable/ic_message_types.xml]: !resolve

Comment (teamagam1) [inline app/src/main/res/drawable/ic_message_types.xml]: !resolved

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (teamagam1) [inline app/src/main/res/layout/activity_main.xml]: !fixed

File: app/src/main/res/values/dimens.xml (hunk 8 above)
Comment (teamagam1) [inline app/src/main/res/values/dimens.xml]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/adapters/MessagesRecyclerViewAdapter.java [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    public int getItemViewType(int position) {
+        return getMessageType(mAccessor.get(position));
+    }
+
+    private static int getMessageType(Message msg) {
+        return sTypeMessageMap.get(msg.getType());
+    }
+
+    protected void bindItemToView(final MessageViewHolder holder) {
+        drawMessageIcon(holder);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/adapters/MessagesRecyclerViewAdapter.java:64]: beautiful

File: app/src/main/java/com/teamagam/gimelgimel/app/view/adapters/MessagesRecyclerViewAdapter.java [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+        drawMessageDate(holder);
+        drawMessageBackground(holder);
+
+        holder.senderTV.setText(holder.item.getSenderId());
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/adapters/MessagesRecyclerViewAdapter.java:68]: you could have done the same here, abstract the little details of binding the sender TextView data with the viewHolder associated string with a simple method extraction

**Pull request merged**